src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.Generic.Tests.cs

```diff
@@ -246,6 +246,35 @@ public void PriorityQueue_DequeueEnqueue(int count)
             AssertExtensions.CollectionEqual(expectedItems, queue.UnorderedItems, EqualityComparer<(TElement, TPriority)>.Default);
         }
 
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void PriorityQueue_Remove_AllElements(int count)
+        {
+            bool result;
+            TElement removedElement;
+            TPriority removedPriority;
+
+            PriorityQueue<TElement, TPriority> queue = CreatePriorityQueue(count, count, out List<(TElement element, TPriority priority)> generatedItems);
+
+            for (int i = count - 1; i >= 0; i--)
+            {
+                (TElement element, TPriority priority) = generatedItems[i];
```

Comment:
>(TElement element, TPriority priority) = generatedItems[i]; [](http://example.com/codeflow?start=16&length=59)

I think this will be better to make it remove from random index instead of always last one. i.e. use `Random.Next(i)` instead of `i`.

Reply:
We generally want to avoid nondeterminism in unit tests. I could try using a `Random` instance with a seed, however the `CreatePriorityQueue` already uses pseudorandomness when populating the heap.

Another thing to note is that the index on `generatedItems` is not a predictor of where the element is located within the heap itself, since it is structured based on the priority values.